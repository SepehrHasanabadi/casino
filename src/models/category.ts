export enum CategoryName {
  top='top',
  new='new',
  slots='slots',
  classic='classic',
  blackjack='blackjack',
  roulette='roulette',
  poker='poker',
  ball='ball',
  virtual='virtual',
  fun='fun'
}

