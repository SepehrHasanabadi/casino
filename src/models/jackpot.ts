/**
 * @description Interface for Jackpot
 * @interface
 * @property {string} game - The id of the Game
 * @property {string} amount - The amount of the jackpot
 */
export interface Jackpot {
  game: string,
  amount: number,
}