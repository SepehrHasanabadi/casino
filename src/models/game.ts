import { CategoryName } from "./category";

/**
 * @description Interface for Category
 * @interface
 * @property {string} name - The name of the Game
 * @property {CategoryName[]} categories - The categories of the Game
 * @property {string} image - The image of the Game
 * @property {string} id - The id of the Game
 */
export interface Game {
  name: string,
  categories: CategoryName[],
  image: string,
  id: string
}