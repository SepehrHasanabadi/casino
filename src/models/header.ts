import { CategoryName } from './category';

export interface HeaderNameInterface {
  top: CategoryName,
  new: CategoryName,
  slots: CategoryName,
  classic: CategoryName,
  blackjack: CategoryName,
  roulette: CategoryName,
  poker: CategoryName,
  other: CategoryName[]
}

export const HeaderName:HeaderNameInterface = {
  top: CategoryName.top,
  new: CategoryName.new,
  slots: CategoryName.slots,
  classic: CategoryName.classic,
  blackjack: CategoryName.blackjack,
  roulette: CategoryName.roulette,
  poker: CategoryName.poker,
  other: [CategoryName.ball, CategoryName.virtual, CategoryName.fun]
}