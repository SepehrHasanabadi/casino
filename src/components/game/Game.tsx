import { CategoryName } from "../../models/category";
import { Game as GameModel } from "../../models/game";
import { Jackpot } from "../../models/jackpot";
import styles from './Game.module.scss'
import GameCard from "./GameCard";

const Game = ({games, active, jackpots}: {
  games?: GameModel[],
  jackpots?: Jackpot[],
  active: CategoryName | CategoryName[],
}) => {
  let categoryGame;
  if (Array.isArray(active)) {
    categoryGame = games?.filter(item => active.includes(item.categories[0]))
  } else {
    categoryGame = games?.filter(item => item.categories.includes(active))
  }
  const jackpotDict: {[key: string]: number} = {};
  jackpots?.forEach(item => jackpotDict[item.game] = item.amount);

  return (
    <div className={styles.grid}>
      {categoryGame && categoryGame.map(item => 
        <GameCard category={active} key={item.id} item={item} jackpot={jackpotDict[item.id]}/>
      )}
    </div>
  );
}

export default Game;