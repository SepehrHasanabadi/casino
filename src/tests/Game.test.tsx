import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect'
import Game from '../components/game/Game';
import { CategoryName } from '../models/category';

it('changes the class when hovered', () => {
  const games = [{
    name: 'test',
    categories: [CategoryName.top, CategoryName.slots],
    image: 'https://image.com',
    id: 'test_id'
  }]
  const jackpots = [{
    game: 'test_id',
    amount: 12000,
  }]
  const component = renderer.create(
    <Game active={CategoryName.top} games={games} jackpots={jackpots}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});