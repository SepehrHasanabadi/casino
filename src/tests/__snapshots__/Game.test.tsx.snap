// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div>
  <div>
    <img
      src="https://image.com"
    />
    <span>
      Â£
      12,000
    </span>
    <div>
      <div>
        test
      </div>
      <div>
        PLay
      </div>
    </div>
  </div>
</div>
`;
